src/swig/CMakeLists.txt

```diff
@@ -58,9 +66,56 @@ if (GUILE_FOUND)
 
   file (RELATIVE_PATH _REL_GUILE_SITE_PATH ${GUILE_ROOT_DIR} ${GUILE_SITE_DIR})
   set (GUILE_SITE_PATH ${_REL_GUILE_SITE_PATH})
-  install (FILES ${CMAKE_CURRENT_BINARY_DIR}/nlopt.scm DESTINATION ${GUILE_SITE_PATH})
+  install (FILES ${CMAKE_CURRENT_BINARY_DIR}/guile/nlopt.scm DESTINATION ${GUILE_SITE_PATH})
 
   file (RELATIVE_PATH _REL_GUILE_EXTENSION_PATH ${GUILE_ROOT_DIR} ${GUILE_EXTENSION_DIR})
   set (GUILE_EXTENSION_PATH ${_REL_GUILE_EXTENSION_PATH})
   install (TARGETS nlopt_guile LIBRARY DESTINATION ${GUILE_EXTENSION_PATH})
 endif ()
+
+
+if (JNI_FOUND AND Java_FOUND AND SWIG_FOUND)
+
+  include (UseJava)
+
+  set (SWIG_MODULE_nlopt_java_EXTRA_DEPS nlopt-java.i generate-cpp)
+  set (CMAKE_SWIG_FLAGS -package nlopt)
+
+  # swig_add_module is deprecated
+  # OUTPUT_DIR is ${CMAKE_CURRENT_BINARY_DIR}/java/ + the -package above (with
+  # any '.' replaced by '/'). It must also match the GLOB in glob_java.cmake.
+  swig_add_library (nlopt_java LANGUAGE java SOURCES nlopt.i
+                               OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/java/nlopt
+                               OUTFILE_DIR ${CMAKE_CURRENT_BINARY_DIR})
+
+  set (CMAKE_SWIG_FLAGS)
+
+  swig_link_libraries (nlopt_java ${nlopt_lib})
+  target_link_libraries (nlopt_java JNI::JNI)
+
+  set_target_properties (nlopt_java PROPERTIES OUTPUT_NAME nloptjni)
+  if (CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
+    set_target_properties (nlopt_java PROPERTIES
+                           COMPILE_OPTIONS "-fno-strict-aliasing")
+  endif ()
+
+  install (TARGETS nlopt_java RUNTIME DESTINATION ${NLOPT_INSTALL_BINDIR}
+                              LIBRARY DESTINATION ${NLOPT_INSTALL_LIBDIR}
+                              ARCHIVE DESTINATION ${NLOPT_INSTALL_LIBDIR})
```

Comment:
that's curious, nloptjni looks still installed in /lib:
```
 Installing: D:/a/nlopt/nlopt/install/lib/nloptjni.dll
```

Reply:
That is because it is of MODULE type. See https://cmake.org/cmake/help/book/mastering-cmake/chapter/Install.html#installing-targets . So if we want to install it to `bin`, we have to set the `LIBRARY DESTINATION` conditionally for Windows.

Reply:
The latest revision (commit 8cd6dad0b8c9fbec9a7282a5531674bf744bd9a3) now sets the destination for `nlopt_java` correctly.